account_move_promissory_note/models/account_move.py

```diff
@@ -0,0 +1,18 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, fields, api
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    payment_ids = fields.Many2many('account.payment', 'account_invoice_payment_rel', 'invoice_id' , 'payment_id')
+    promissory_note_number = fields.Char(compute='_compute_promissory_note_number', store=True)
+
+    @api.depends('payment_ids', 'payment_ids.promissory_note_number')
+    def _compute_promissory_note_number(self):
+        for record in self:
+            record.promissory_note_number = ''
+            for payment in record.payment_ids.filtered(lambda x: x.promissory_note and x.promissory_note_number):
+                record.promissory_note_number += '%s ' % (payment.promissory_note_number)
```

Comment:
In a single line (not tested):
```suggestion
            record.promissory_note_number = " ".join(record.payment_ids.mapped("promissory_note_number"))
```

Reply:
"filtered(lambda x: x.promissory_note_number)" its mandatory because otherwise it generates boolean error